bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/AzureBlobWatcherHandler.java

```diff
@@ -0,0 +1,134 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.folderwatcher.internal.api.AzureActions;
+import org.openhab.binding.folderwatcher.internal.common.WatcherCommon;
+import org.openhab.binding.folderwatcher.internal.config.AzureBlobWatcherConfiguration;
+import org.openhab.core.OpenHAB;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AzureBlobWatcherHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Alexandr Salamatov - Initial contribution
+ */
+@NonNullByDefault
+public class AzureBlobWatcherHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(AzureBlobWatcherHandler.class);
+    private AzureBlobWatcherConfiguration config = new AzureBlobWatcherConfiguration();
+    private File currentBlobListingFile = new File(OpenHAB.getUserDataFolder() + File.separator + "AzureBlob"
+            + File.separator + thing.getUID().getAsString().replace(':', '_') + ".data");
+    private @Nullable ScheduledFuture<?> executionJob;
+    private List<String> previousBlobListing = new ArrayList<>();
+    private HttpClientFactory httpClientFactory;
+    private @Nullable AzureActions azure;
+
+    public AzureBlobWatcherHandler(Thing thing, HttpClientFactory httpClientFactory) {
+        super(thing);
+        this.httpClientFactory = httpClientFactory;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Channel {} triggered with command {}", channelUID.getId(), command);
+        if (command instanceof RefreshType) {
+            refreshAzureBlobInformation();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(AzureBlobWatcherConfiguration.class);
+
+        if (config.azureAnonymous) {
+            azure = new AzureActions(httpClientFactory, config.azureAccountName, config.azureContainerName);
+        } else {
+            azure = new AzureActions(httpClientFactory, config.azureAccountName, config.azureContainerName,
+                    config.azureAccessKey);
+        }
+        updateStatus(ThingStatus.UNKNOWN);
+        if (config.pollIntervalAzure > 0) {
+            executionJob = scheduler.scheduleWithFixedDelay(this::refreshAzureBlobInformation, 0,
+                    config.pollIntervalAzure, TimeUnit.SECONDS);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Polling interval must be greater then 0 seconds");
+            return;
+        }
```

Comment:
Improved implementation (might need spotless, as i wrote it here in github)
```suggestion
        if (config.azureAccountName.isBlank() || config.azureContainerName.isBlank()) {
                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
                    "Account name or container configuration is invalid");
                return;
        } else if (config.pollIntervalAzure == 0) {
                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
                    "Polling interval must be greater then 0 seconds");
                return;
        }
        
        if (config.azureAnonymous) {
            azure = new AzureActions(httpClientFactory, config.azureAccountName, config.azureContainerName);
        } else {
            azure = new AzureActions(httpClientFactory, config.azureAccountName, config.azureContainerName,
                    config.azureAccessKey);
        }

        updateStatus(ThingStatus.UNKNOWN);
        executionJob = scheduler.scheduleWithFixedDelay(this::refreshAzureBlobInformation, 0,
                    config.pollIntervalAzure, TimeUnit.SECONDS);
```